state/protocol/badger/mutator.go

```diff
@@ -449,14 +438,27 @@ func (m *FollowerState) insert(ctx context.Context, candidate *flow.Block, last
 			return fmt.Errorf("could not index new block: %w", err)
 		}
 
+		// if applicable, flag that epoch fallback has been triggered
+		if insertingBlockTriggersEpochFallback {
+			err = transaction.WithTx(operation.SetEpochEmergencyFallbackTriggered(blockID))(tx)
+			if err != nil {
+				return fmt.Errorf("could not set epoch fallback triggered flag: %w", err)
+			}
+		}
+
 		// apply any optional DB operations from service events
-		for _, apply := range ops {
+		for _, apply := range dbUpdates {
 			err := apply(tx)
 			if err != nil {
 				return fmt.Errorf("could not apply operation: %w", err)
 			}
 		}
 
+		// emit protocol events
```

Comment:
Follow-up on our discussion: 
* please include TODO's here and create an issue to update the implementation of the service events to be completely asynchronous (with recovery logic where needed)

Reply:
Created issue: https://github.com/dapperlabs/flow-go/issues/6317